encoding/ssz/query/analyzer.go

```diff
@@ -0,0 +1,235 @@
+package query
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+const (
+	offsetBytes = 4
+
+	// sszMaxTag specifies the maximum capacity of a variable-sized collection, like an SSZ List.
+	sszMaxTag = "ssz-max"
+
+	// sszSizeTag specifies the length of a fixed-sized collection, like an SSZ Vector.
+	// A wildcard ('?') indicates that the dimension is variable-sized (a List).
+	sszSizeTag = "ssz-size"
+)
+
+// AnalyzeSSZInfo analyzes given object and returns its SSZ information.
+func AnalyzeSSZInfo(obj any) (*sszInfo, error) {
+	value := dereferencePointer(obj)
+
+	info, err := analyzeType(value.Type(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("analyze type %s: %w", value.Type().Name(), err)
+	}
+
+	return info, nil
+}
+
+// analyzeType is an entry point that inspects a reflect.Type and computes its SSZ layout information.
+func analyzeType(typ reflect.Type, tag *reflect.StructTag) (*sszInfo, error) {
+	switch typ.Kind() {
+	// Basic types (e.g., uintN where N is 8, 16, 32, 64)
+	// NOTE: uint128 and uint256 are represented as []byte in Go,
+	// so we handle them as slices. See `analyzeHomogeneousColType`.
+	case reflect.Uint64, reflect.Uint32, reflect.Uint16, reflect.Uint8, reflect.Bool:
+		return analyzeBasicType(typ)
+
+	case reflect.Slice:
+		return analyzeHomogeneousColType(typ, tag)
+
+	case reflect.Struct:
+		return analyzeContainerType(typ)
+
+	case reflect.Ptr:
+		// Dereference pointer types.
+		return analyzeType(typ.Elem(), tag)
+
+	default:
+		return nil, fmt.Errorf("unsupported type for SSZ calculation: %v", typ.Kind())
+	}
+}
+
+// analyzeBasicType analyzes SSZ basic types (uintN, bool) and returns its info.
+func analyzeBasicType(typ reflect.Type) (*sszInfo, error) {
+	sszInfo := &sszInfo{
+		typ: typ,
+
+		// Every basic type is fixed-size and not variable.
+		isVariable: false,
+	}
+
+	switch typ.Kind() {
+	case reflect.Uint64:
+		sszInfo.sszType = UintN
+		sszInfo.fixedSize = 8
+	case reflect.Uint32:
+		sszInfo.sszType = UintN
+		sszInfo.fixedSize = 4
+	case reflect.Uint16:
+		sszInfo.sszType = UintN
+		sszInfo.fixedSize = 2
+	case reflect.Uint8:
+		sszInfo.sszType = UintN
+		sszInfo.fixedSize = 1
+	case reflect.Bool:
+		sszInfo.sszType = Boolean
+		sszInfo.fixedSize = 1
+	default:
+		return nil, fmt.Errorf("unsupported basic type for SSZ calculation: %v", typ.Kind())
```

Comment:
```suggestion
		return nil, fmt.Errorf("unsupported basic type %v for SSZ calculation", typ.Kind())
```
Errors in Go shouldn't contain punctuation, especially `:` because it separates errors

Reply:
Done for other place in this file as well.